ngraph/core/src/op/experimental_detectron_prior_grid_generator.cpp

```diff
@@ -0,0 +1,157 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+
+#include "itt.hpp"
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/op/experimental_detectron_prior_grid_generator.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+using GridGenerator = op::v6::ExperimentalDetectronPriorGridGenerator;
+
+NGRAPH_RTTI_DEFINITION(GridGenerator, "ExperimentalDetectronPriorGridGenerator", 6);
+
+GridGenerator::ExperimentalDetectronPriorGridGenerator(const Output<Node>& priors,
+                                                       const Output<Node>& feature_map,
+                                                       const Output<Node>& im_data,
+                                                       const Attributes& attrs)
+    : Op({priors, feature_map, im_data})
+    , m_attrs(attrs)
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::ExperimentalDetectronPriorGridGenerator::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("flatten", m_attrs.flatten);
+    visitor.on_attribute("h", m_attrs.h);
+    visitor.on_attribute("w", m_attrs.w);
+    visitor.on_attribute("stride_x", m_attrs.stride_x);
+    visitor.on_attribute("stride_y", m_attrs.stride_y);
+    return true;
+}
+
+shared_ptr<Node> GridGenerator::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+
+    return make_shared<op::v6::ExperimentalDetectronPriorGridGenerator>(
+        new_args.at(0), new_args.at(1), new_args.at(2), m_attrs);
+}
+
+static constexpr size_t priors_port = 0;
```

Comment:
Looks like we don't need to have this global constant indexes if `validate()` method has as arguments `priors_shape`, `featmap_shape ` and `im_data_shape`.

Moreover why do we need a separate `validate()` method if we use it only in one place.

Reply:
If we put the body of `validate()` into `validate_and_infer_type()`,  we get long function, and this long function is hard to read.